packages/lodestar/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -0,0 +1,310 @@
+import bls from "@chainsafe/bls";
+import {ForkName, SLOTS_PER_EPOCH} from "@chainsafe/lodestar-params";
+import {allForks, altair, Slot, ssz, ValidatorIndex} from "@chainsafe/lodestar-types";
+import {
+  CachedBeaconState,
+  computeEpochAtSlot,
+  phase0,
+  zipIndexesCommitteeBits,
+} from "@chainsafe/lodestar-beacon-state-transition";
+import {List, readonlyValues, toHexString} from "@chainsafe/ssz";
+import {MapDef} from "../../util/map";
+import {pruneBySlot} from "./utils";
+import {InsertOutcome, OpPoolError, OpPoolErrorCode} from "./types";
+
+type DataRootHex = string;
+/**
+ * Maintain a pool of aggregated attestations. Attestations can be retrieved for inclusion in a block
+ * or api. The returned attestations are aggregated to maximise the number of validators that can be
+ * included.
+ * Note that we want to remove attestations with attesters that were included in the chain.
+ */
+export class AggregatedAttestationPool {
+  private readonly attestationGroupByDataHashBySlot = new MapDef<
+    phase0.Slot,
+    Map<DataRootHex, MatchingDataAttestationGroup>
+  >(() => new Map<DataRootHex, MatchingDataAttestationGroup>());
+  private lowestPermissibleSlot = 0;
+
+  add(attestation: phase0.Attestation, attestingIndices: ValidatorIndex[], committee: ValidatorIndex[]): InsertOutcome {
+    const slot = attestation.data.slot;
+    const lowestPermissibleSlot = this.lowestPermissibleSlot;
+
+    // Reject any attestations that are too old.
+    if (slot < lowestPermissibleSlot) {
+      throw new OpPoolError({code: OpPoolErrorCode.SLOT_TOO_LOW, slot, lowestPermissibleSlot});
```

Comment:
Seeng that this error happens frequently on the attestation pool, should it just be an InsertOutcome?

Reply:
I don't see any of it for this pool but think it's good to do that since the unaggregated pool throws a lot of errors like that. So we should return `AlreadyKnown`?